angr/analyses/proximity_graph.py

```diff
@@ -180,6 +180,33 @@ def __init__(
 
         self._work()
 
+    def _condense_blank_nodes(self, graph):
+        nodes = list(graph.nodes)
+        blank_nodes = []
+
+        for node in nodes:
+            if isinstance(node, BaseProxiNode) and node.type_ == ProxiNodeTypes.Empty:
+                blank_nodes.append(node)
+            else:
+                if blank_nodes:
+                    self._merge_nodes(graph, blank_nodes)
+                    blank_nodes = []
+
+        if blank_nodes:
+            self._merge_nodes(graph, blank_nodes)
+
+    def _merge_nodes(self, graph, nodes):
```

Comment:
Would you be able to type annotate `nodes` in this context?

Reply:
Ah yeah! My bad, I normally have alerts on for type annotations not being added but I disabled them as a bunch of stuff didn't have them, added now :-)